### PR TITLE
sdl_mixer: add v2.8.0 as a new subdir

### PR DESCRIPTION
**Pull request by valgur**
The existing recipe is quite outdated. The project has switched over to using CMake.

This also effectively migrates the package to Conan v2.

### CODE REVIEW

**Comment by RubenRBS** [comment]
Rebuilding the missing dependency now :)

File: recipes/sdl_mixer/cmake/conanfile.py
Comment (valgur) [reply recipes/sdl_mixer/cmake/conanfile.py]: Indeed. Thank you!

Comment (petitlapin) [reply recipes/sdl_mixer/cmake/conanfile.py]: About the remaining issues, do you have an idea of what goes wrong? I don't know much conan but I can try to take a look (with no guarantee) when I have some spare time

Comment (valgur) [reply recipes/sdl_mixer/cmake/conanfile.py]: @petitlapin I looked into it and I think it's caused by this issue: #22170

File: recipes/sdl_mixer/cmake/test_package/test_package.c
Comment (petitlapin) [inline recipes/sdl_mixer/cmake/test_package/test_package.c:4]: I think it would be better to include SDL2 in the path to ensure we look for SDL2 (and not getting SDL by mistake)
```suggestion
#include <SDL2/SDL.h>
#include <SDL2/SDL_mixer.h>
```

File: recipes/sdl_mixer/cmake/conanfile.py
Comment (petitlapin) [inline recipes/sdl_mixer/cmake/conanfile.py:192]: I don't know if it is a good track to follow to try to fix the error but in sdl_ttf: https://github.com/conan-io/conan-center-index/blob/master/recipes/sdl_ttf/all/conanfile.py#L115
```
suffix = "-static" if Version(self.version) >= "2.20.0" and not self.options.shared else ""
self.cpp_info.components["_sdl2_ttf"].set_property("cmake_target_name",f"SDL2_ttf::SDL2_ttf{suffix}")
```
and in the test [CMakeLists.txt](https://github.com/conan-io/conan-center-index/blob/master/recipes/sdl_ttf/all/test_package/CMakeLists.txt#L7C1-L11C8) they use:
```
if(TARGET SDL2_ttf::SDL2_ttf)
    target_link_libraries(${PROJECT_NAME} PRIVATE SDL2_ttf::SDL2_ttf)
else()
    target_link_libraries(${PROJECT_NAME} PRIVATE SDL2_ttf::SDL2_ttf-static)
endif()
```

Comment (valgur) [reply recipes/sdl_mixer/cmake/conanfile.py:192]: Yes, it should. That's what the targets exported by the `sdl_mixer` project look like as well: https://github.com/libsdl-org/SDL_mixer/blob/release-2.6.3/CMakeLists.txt#L164-L172

File: recipes/sdl_mixer/cmake/conanfile.py
Comment (SpaceIm) [inline recipes/sdl_mixer/cmake/conanfile.py:166]:
```suggestion
        deps.set_property("libmodplug", "cmake_file_name", "modplug")
        deps.set_property("libmodplug", "cmake_target_name", "modplug::modplug")
```

Here is your issue, https://github.com/conan-io/conan-center-index/pull/22170 is irrelevant.

Comment (valgur) [reply recipes/sdl_mixer/cmake/conanfile.py:166]: Thanks!

File: recipes/sdl_mixer/cmake/conanfile.py
Comment (petitlapin) [inline recipes/sdl_mixer/cmake/conanfile.py:188]: On Windows, it seems to be `self, os.path.join(self.package_folder, "cmake")`. The suggestion is not tested:
```suggestion
        if self.settings.os == "Windows":
            rmdir(self, os.path.join(self.package_folder, "cmake"))
        else:
            rmdir(self, os.path.join(self.package_folder, "lib", "cmake"))
```

From the logs:
Linux:
> -- Install configuration: "Release"
> -- Installing: /home/conan/w/prod-v2/bsr/70187/fbaed/p/b/sdl_md2106973f57d3/p/lib/libSDL2_mixer.a
> -- Installing: /home/conan/w/prod-v2/bsr/70187/fbaed/p/b/sdl_md2106973f57d3/p/include/SDL2/SDL_mixer.h
> -- Installing: /home/conan/w/prod-v2/bsr/70187/fbaed/p/b/sdl_md2106973f57d3/p/lib/cmake/SDL2_mixer/SDL2_mixerConfig.cmake


Windows:
> -- Installing: C:/J2/w/prod-v1/bsr@2/84698/ceffe/.conan/data/sdl_mixer/2.6.3/_/_/package/51188b1528b7de4f0f4ac2754c459259f18a23cd/lib/SDL2_mixer-static.lib
> -- Installing: C:/J2/w/prod-v1/bsr@2/84698/ceffe/.conan/data/sdl_mixer/2.6.3/_/_/package/51188b1528b7de4f0f4ac2754c459259f18a23cd/include/SDL2/SDL_mixer.h
> -- Installing: C:/J2/w/prod-v1/bsr@2/84698/ceffe/.conan/data/sdl_mixer/2.6.3/_/_/package/51188b1528b7de4f0f4ac2754c459259f18a23cd/cmake/SDL2_mixerConfig.cmake

After checking, the code in SDL_mixer that set this variable for Windows is: https://github.com/libsdl-org/SDL_mixer/blob/SDL2/CMakeLists.txt#L910. I'm not sure of the reason but if there is a if in the code, there should be a reason so the workaround in conan may make sense...

**Comment by petitlapin** [comment]
For the Linux error, I found https://github.com/conan-io/conan-center-index/pull/22081 which removed the dependency a few days ago for xorg. I don't know if pulseaudio should be rebuilt with this version to remove the dependency?

**Comment by Ahajha** [comment]
> For the Linux error, I found #22081 which removed the dependency a few days ago for xorg. I don't know if pulseaudio should be rebuilt with this version to remove the dependency?

I'm running into the same issue in CPython. It's possible a bunch of dependencies will need to be rebuilt.

**Comment by Croydon** [comment]
Pulseaudio rebuild is happing in  #22241, awaiting approval

File: recipes/sdl_mixer/cmake/conanfile.py
Comment (Croydon) [inline recipes/sdl_mixer/cmake/conanfile.py:1]: How much of the recipe itself did you change? Is it worth the file split?

Comment (valgur) [reply recipes/sdl_mixer/cmake/conanfile.py:1]: The project changed its CMake config and dependencies substantially since the current v2.0.4.
Cramming it all together seemed like a worse option with no real upside.
Here's a PR to migrate the existing recipe: #21782

**Comment by petitlapin** [comment]
According to the [changelog](https://github.com/libsdl-org/SDL_mixer/releases/tag/release-2.8.0):
```
Use libxmp instead of modplug as the default backend for MOD music
To use libmodplug instead, configure using --enable-music-mod-modplug --disable-music-mod-xmp
```

I guess the recipe should either use libxmp or force the use of modplug.

**Comment by petitlapin** [comment]
hi, is there a reason to not merge it?

**Review by zaslavskyv — APPROVED** [review]

File: recipes/sdl_mixer/cmake/conanfile.py
Comment (Croydon) [inline recipes/sdl_mixer/cmake/conanfile.py:129]:
```suggestion
```

Seems unnecessary as Conan does this automatically

File: recipes/sdl_mixer/cmake/conanfile.py
Comment (Croydon) [inline recipes/sdl_mixer/cmake/conanfile.py:136]: Context/Source?

File: recipes/sdl_mixer/cmake/conanfile.py
Comment (Croydon) [inline recipes/sdl_mixer/cmake/conanfile.py:131]: This also manipulates `SDL2MIXER_DEBUG_POSTFIX` for some reason, like the lines a few lines down

Based on the version check this seems deleteable as this recipe only manages newer versions. But please clarify in general why this postfix is deleted by us conditionally 🤔

File: recipes/sdl_mixer/cmake/conanfile.py
Comment (Croydon) [inline recipes/sdl_mixer/cmake/conanfile.py:172]: Was this broken in previous recipe revisions? Seemingly it got supported 🤔

**Comment by Croydon** [comment]
Please edit the pull request title regarding the version, so that the merge commit message will be correct

File: recipes/sdl_mixer/cmake/conanfile.py
Comment (valgur) [reply recipes/sdl_mixer/cmake/conanfile.py:129]: Thanks! The project uses a new enough CMake policy version (3.16) and indeed does not require the use of `cache_variables`.

File: recipes/sdl_mixer/cmake/conanfile.py
Comment (valgur) [reply recipes/sdl_mixer/cmake/conanfile.py:172]: No idea. I'm not going to add it in this PR either way.

**Review by Croydon — APPROVED** [review]

**Comment by Croydon** [comment]
@RubenRBS Can we please get a review? 🙂

File: recipes/sdl_mixer/cmake/test_v1_package/CMakeLists.txt
Comment (memsharded) [inline recipes/sdl_mixer/cmake/test_v1_package/CMakeLists.txt:1]: test_v1_package are no longer necessary, it is better to not add them, specially for new recipes (this under ``cmake`` is a new recipe)

Comment (memsharded) [reply recipes/sdl_mixer/cmake/test_v1_package/CMakeLists.txt:1]: No need to change it now, but just FYI. Less is more, specially when the backlog in ConanCenter is so big.

**Review by memsharded — APPROVED** [review]

**Pull request merged**
